ckanext/unhcr/validators.py

```diff
@@ -212,3 +212,19 @@ def activity_type_exists(activity_type):
     if activity_type in _object_id_validators:
         return activity_type
     return validators.activity_type_exists(activity_type)
+
+
+# Collaborators
+
+def owner_org_validator(key, data, errors, context):
+    user = context.get('user')
+    package = context.get('package')
+    action = toolkit.get_action('dataset_collaborator_list_for_user')
+    if user and action and package:
+        datasets = action(context, {'id': user})
+        if package.id in [d['dataset_id'] for d in datasets]:
+            if data.get(key) == package.owner_org:
+                return
```

Comment:
We still want the value to get validated using the core logic

```suggestion
                return validators.owner_org_validator(key, data, errors, context)
```

Reply:
If the user is a collaborator on a dataset and we fall back to `validators.owner_org_validator()` here, we will then check if the user is in the org and throw `Invalid(_('You cannot add a dataset to this organization'))`. One of the reasons for implementing a custom validator here is to bypass that check. I've proposed a solution in 739cdde although I'm not mad keen on relying on the exception message string.